EMS/helpers/src/File/CsvFile.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EMS\Helpers\File;
+
+/**
+ * @implements \IteratorAggregate<int, mixed>
+ */
+class CsvFile implements \Countable, \IteratorAggregate
+{
+    private const DEFAULT_DELIMITER = ',';
+
+    public function __construct(
+        private readonly string $filename,
+        private readonly ?string $delimiter = null,
+        private readonly ?string $encoding = null,
+    ) {
+    }
+
+    /**
+     * @yield array<int, mixed>
+     */
+    public function getIterator(): \Generator
+    {
+        $handle = $this->getHandle();
+        while (($row = \fgetcsv($handle, 2000, $this->getDelimiter())) !== false) {
+            if ($this->encoding) {
+                $row = \array_map([$this, 'convertEncoding'], $row);
+            }
+            yield $row;
+        }
+        \fclose($handle);
+    }
+
+    public function count(): int
+    {
+        $count = 0;
+        $handle = $this->getHandle();
+        while (false !== \fgetcsv($handle, 2000, $this->getDelimiter())) {
+            ++$count;
+        }
+        \fclose($handle);
+
+        return $count;
+    }
+
+    /**
+     * @return resource
+     */
+    private function getHandle()
+    {
+        if (false === $handle = \fopen($this->filename, 'rb')) {
+            throw new \RuntimeException(\sprintf('Could not open "%s"', $this->filename));
+        }
+
+        return $handle;
+    }
+
+    private function getDelimiter(): string
+    {
+        return $this->delimiter ?? self::DEFAULT_DELIMITER;
+    }
+
```

Comment:
useless to me

Reply:
Fixed https://github.com/ems-project/elasticms/pull/1056/commits/0db327e01b0755d8448261ae86ea17fc2365b857